### PR TITLE
Static execution adapter

### PR DESCRIPTION
**Pull request by peterschwarz**
First pass implementation of the static execution adapter and all of the necessary changes to make it work.

Replaces #32

### CODE REVIEW

File: libtransact/src/context/manager.rs [hunk 1]
```diff
@@ -283,6 +284,139 @@ impl<
...
+    }
+
+    impl<K, V> ContextManager<K, V>
+    where
+        K: Send + Hash + Eq + Clone + Debug,
+        V: Send + Clone + Debug,
+    {
+        /// Constructs a new Context Manager around a given state Read.
+        ///
+        /// The Read defines the state on which the context build.
```
Comment (jsmitchell) [inline libtransact/src/context/manager.rs]: s/context build/context is built/ ?

File: libtransact/src/context/manager.rs [hunk 2]
```diff
@@ -283,6 +284,139 @@ impl<
...
+        }
+
+        pub fn add_event(
+            &self,
+            context_id: &ContextId,
+            event: Event,
+        ) -> Result<(), ContextManagerError> {
+            self.internal_manager
+                .lock()
+                .expect("Lock in the add_event was poisoned")
```
Comment (jsmitchell) [inline libtransact/src/context/manager.rs]: Half of these are "Lock in the" and half are "Lock in"

File: libtransact/src/execution/adapter/error.rs [hunk 3]
```diff
@@ -16,16 +16,51 @@
  */
 
 use crate::transaction::TransactionPair;
+use std::{error::Error, fmt};
```
Comment (agunde406) [inline libtransact/src/execution/adapter/error.rs]: Typo in commit message, corret -> correct

Comment (peterschwarz) [reply libtransact/src/execution/adapter/error.rs]: fixed

File: libtransact/src/context/manager.rs (hunk 2 above)
Comment (peterschwarz) [reply libtransact/src/context/manager.rs]: fixed

File: libtransact/src/context/manager.rs (hunk 1 above)
Comment (peterschwarz) [reply libtransact/src/context/manager.rs]: fixed.

**Pull request closed without merging**